packages/dds/tree/src/test/shared-tree/sharedTree.bench.ts

```diff
@@ -0,0 +1,877 @@
+	});
+	const testSchemaMap: Map<TreeSchemaIdentifier, NamedTreeSchema> = new Map();
+	testSchemaMap.set(rootSchemaName, testRootNodeSchema);
+	testSchemaMap.set(dataSchema.name, dataSchema);
+	return {
+		treeSchema: testSchemaMap,
+		globalFieldSchema: new Map([]),
+	};
+}
+
+const dataSchema = namedTreeSchema({
+	name: brand("DataSchema"),
+	localFields: {
+		foo: fieldSchema(FieldKinds.optional),
+	},
+	value: ValueSchema.Number,
+});
+
+// number of nodes in test for wide trees
+const nodesCountWide = [
+	[1, BenchmarkType.Measurement],
+	[100, BenchmarkType.Perspective],
+	[500, BenchmarkType.Measurement],
+];
+// number of nodes in test for deep trees
+const nodesCountDeep = [
+	[1, BenchmarkType.Measurement],
+	[10, BenchmarkType.Perspective],
+	[100, BenchmarkType.Measurement],
+];
+
+const rootFieldSchema = fieldSchema(FieldKinds.value);
+const globalFieldSchema = fieldSchema(FieldKinds.value);
+const rootNodeSchema = namedTreeSchema({
+	name: brand("TestValue"),
+	localFields: {
+		optionalChild: fieldSchema(FieldKinds.optional, [brand("TestValue")]),
+	},
+	extraLocalFields: fieldSchema(FieldKinds.sequence),
+	globalFields: [globalFieldKey],
+});
+const testSchema: SchemaData = {
+	treeSchema: new Map([[rootNodeSchema.name, rootNodeSchema]]),
+	globalFieldSchema: new Map([
+		[rootFieldKey, rootFieldSchema],
+		[globalFieldKey, globalFieldSchema],
+	]),
+};
+
+// TODO: Once the "BatchTooLarge" error is no longer an issue, extend tests for larger trees.
+describe("SharedTree benchmarks", () => {
+	it("can move nodes across fields", async () => {
+		const [provider, trees] = await createSharedTrees(
+			getTestSchema(FieldKinds.sequence),
+			[{ type: rootSchemaName }],
+			1,
+		);
+		const tree = trees[0];
+		insertNodesToEditableTree(tree, 4, TreeShape.Wide, TestPrimitives.Float);
+	});
+	describe("Direct JS Object", () => {
+		for (let dataType = 0 as TestPrimitives; dataType <= 4; dataType++) {
+			for (const [i, benchmarkType] of nodesCountDeep) {
+				let tree: Jsonable;
+				benchmark({
+					type: benchmarkType,
+					title: `Deep Tree as JS Object (${TestPrimitives[dataType]}): reads with ${i} nodes`,
+					before: async () => {
+						tree = getTestTreeAsJSObject(i, TreeShape.Deep, dataType);
+					},
+					benchmarkFn: () => {
+						readTreeAsJSObject(tree);
+					},
+				});
+			}
+			for (const [i, benchmarkType] of nodesCountWide) {
+				let tree: Jsonable;
+				benchmark({
+					type: benchmarkType,
+					title: `Wide Tree as JS Object (${TestPrimitives[dataType]}): reads with ${i} nodes`,
+					before: async () => {
+						tree = getTestTreeAsJSObject(i, TreeShape.Wide, dataType);
+					},
+					benchmarkFn: () => {
+						readTreeAsJSObject(tree);
+					},
+				});
+			}
+			for (const [i, benchmarkType] of nodesCountDeep) {
+				let tree: Jsonable;
+				benchmark({
+					type: benchmarkType,
+					title: `Deep Tree as JS Object (${TestPrimitives[dataType]}): writes with ${i} nodes`,
+					before: async () => {},
```

Comment:
the before function is optional, so if you don't need it you should omit it.

Reply:
removed unused before function